### DISCUSSION TITLE
Cannot Update Table

### DISCUSSION THREAD

**Discussion by fadethegap** [supabase/supabase#4731] [Questions] [answered]
I am using the SaaS Starter project with Next.js, and when a user is created using Name/Email/Password, their Name is not saving. When I use Facebook or Google to create a user it works fine.

I referenced this article: [https://github.com/supabase/supabase/discussions/3476](url), and added "with check" to the Update but I'm still getting a 404 error.

I tried running the update with a timeout as well just for the heck of it.

if (createdUser) {
        console.log(createdUser)
        await updateUserName(createdUser, name);
        setNewUser(createdUser);
        setTimeout(() => {
          updateName(createdUser, name)
        }, 2000)
        console.log("Name:", name)
        console.log(createdUser)
      }

And I get the following:

Console:
signup.tsx:119 
{id: '6f66d501-a068-4d31-969a-5e946232e22f', aud: 'authenticated', role: 'authenticated', email: 'test@testuser.com', phone: '', …}
browser-ponyfill.js:518 PATCH https://myURL.supabase.co/rest/v1/users?id=eq.6f66d501-a068-4d31-969a-5e946232e22f 404

If I confirm the account and login, I can call the update no problem. 

Any ideas on what I'm doing wrong?

**Comment by GaryAustin1** [comment, marked as answer]
You say you added the with check policy to update, did you also add the using clause?  Using is required with check is optional.
Edit:  You really should add a bit more to show the supabase calls your function calls are making.

**Reply by fadethegap** [reply]
Thank you @GaryAustin1 
The USING was in the initial policy. Per your recommendation I added the WITH CHECK.
![Screen Shot 2021-12-29 at 11 25 45 AM](https://user-images.githubusercontent.com/47059192/147683147-e99ded9c-20b1-402c-ad17-06d1554b4e82.png)

**Reply by GaryAustin1** [reply]
If you had that correct (just having using or both for update is OK) then I assume your problem still exists and you'll likely need to provide more info on your code as you only refer to some project and have no detail on the supabase calls.

**Reply by fadethegap** [reply]
There is a signup form:
![Screen Shot 2021-12-29 at 11 46 37 AM](https://user-images.githubusercontent.com/47059192/147685125-bb207eff-273c-40e9-b8f2-b2225b773faf.png)

Once submitted calls SignUp
![Screen Shot 2021-12-29 at 11 46 55 AM](https://user-images.githubusercontent.com/47059192/147685194-a66008ca-5484-4ca1-a5d5-36c0412f3286.png)

The updateUserName function is throwing the error.
![Screen Shot 2021-12-29 at 11 47 12 AM](https://user-images.githubusercontent.com/47059192/147685255-4186d7b7-8e6b-46ff-8b38-436021801c9e.png)

In the Console the error is:
![Screen Shot 2021-12-29 at 11 53 16 AM](https://user-images.githubusercontent.com/47059192/147685691-ee8c111d-c890-40a0-91bf-45178caebd72.png)

And the Network:
![Screen Shot 2021-12-29 at 11 55 30 AM](https://user-images.githubusercontent.com/47059192/147685860-8ef4360f-f8d6-4f21-a07c-30fdaca657b8.png)
![Screen Shot 2021-12-29 at 11 56 08 AM](https://user-images.githubusercontent.com/47059192/147685874-a73fbd8a-ab9f-4f6d-9199-264abbb991ec.png)

The Auth --> Users table - a row is inserted
![Screen Shot 2021-12-29 at 11 53 49 AM](https://user-images.githubusercontent.com/47059192/147685923-6dc24bf2-1489-443c-b71b-45d8f96daf59.png)

And in the Users table a row is inserted, but the full_name field is not updated.
![Screen Shot 2021-12-29 at 11 54 08 AM](https://user-images.githubusercontent.com/47059192/147686023-2b4aa4a3-27a3-4ce2-81bb-0e689ec593d5.png)

Once I use the user credentials to sign in, I can call the updateUserName function on a button click and it works fine.
![Screen Shot 2021-12-29 at 12 00 56 PM](https://user-images.githubusercontent.com/47059192/147686173-6ac321ad-a1dd-44ae-871d-bc18b2254a16.png)

**Reply by GaryAustin1** [reply]
There needs to be a user session to update the users table if you have RLS with uid()=id.
Check this part on email confirmations in the Supabase signUp API guide and what signUp does.

By default, the user will need to verify their email address before logging in. If you would like to change this, you can disable "Email Confirmations" by going to Authentication -> Settings on app.supabase.io
If "Email Confirmations" is turned on, a user is returned but session will be null
If "Email Confirmations" is turned off, both a user and a session will be returned

**Reply by fadethegap** [reply]
Thank you Gary! I had read through that but it didn't register.
